src/test/scala/nl/hugo/redbook/ch7/ExecutorServiceDecorator.scala

```diff
@@ -0,0 +1,11 @@
+package nl.hugo.redbook.ch7
+
+import java.util.concurrent.{ ExecutorService, ThreadPoolExecutor }
+
+object ExecutorServiceDecorator {
+
+  implicit class ExectorServiceOps(es: ExecutorService) {
```

Comment:
Adding `extends AnyVal` optimizes away the allocation of an `ExecutorServiceOps` instance.